src/query_engine/query_engine.cpp

```diff
@@ -1040,32 +1036,29 @@ silo::QueryResult silo::execute_query(
             << " microseconds\n";
 
    {
-      BlockTimer timer(query_result.action_time);
+      BlockTimer const timer(query_result.action_time);
       const auto& action = doc["action"];
       assert(action.HasMember("type"));
       assert(action["type"].IsString());
       const auto& action_type = action["type"].GetString();
 
       if (action.HasMember("groupByFields")) {
          assert(action["groupByFields"].IsArray());
-         std::vector<std::string> groupByFields;
-         for (const auto& it : action["groupByFields"].GetArray()) {
-            groupByFields.emplace_back(it.GetString());
-         }
-         if (strcmp(action_type, "Aggregated") == 0) {
-         } else if (strcmp(action_type, "List") == 0) {
-         } else if (strcmp(action_type, "Mutations") == 0) {
-         } else {
-            query_result.queryResult = response::ErrorResult{
-               "Unknown action", std::string(action_type) + " is not a valid action"};
+         std::vector<std::string> group_by_fields;
+         for (const auto& field : action["groupByFields"].GetArray()) {
+            group_by_fields.emplace_back(field.GetString());
          }
+
+         query_result.queryResult = response::ErrorResult{
+            "Unknown action", std::string(action_type) + " is not a valid action"};
```

Comment:
Let's rather return a "groupByFields is not properly implemented yet" exception instead. Your change changes the logic from "silently ignore valid cases" to "everything is invalid".